src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
```

Comment:
please rename it like `array` -> `dataContainer`. It is quite confusing what array means and what it is doing. The variable names should explain for what purpose we need them (like description). It is the same like we have the Dog class and instead of `int paws` we will use `int nums` and for what that nums we need we can understand only when we get deep dive into the code. I would recommend to read `Clean code, Rober Martin` this point is well described there, u can read not from the scratch but the point what u interested about. Like u r not sure how to write some point of code (about stylistics) u can just open in book and read about that